src/tree.rs

```diff
@@ -95,6 +95,16 @@ impl From<(incrementalmerkletree::Position, Vec<MerkleHashOrchard>)> for MerkleP
     }
 }
 
+impl From<incrementalmerkletree::MerklePath<MerkleHashOrchard, 32>> for MerklePath {
+    fn from(path: incrementalmerkletree::MerklePath<MerkleHashOrchard, 32>) -> Self {
```

Comment:
Does this work?
```suggestion
impl From<incrementalmerkletree::MerklePath<MerkleHashOrchard, MERKLE_DEPTH_ORCHARD>> for MerklePath {
    fn from(path: incrementalmerkletree::MerklePath<MerkleHashOrchard, MERKLE_DEPTH_ORCHARD>) -> Self {
```

Reply:
Doesn't work; `MERKLE_DEPTH_ORCHARD` is a `usize`.

Reply:
We should figure out a way to make (something like) this work, even if it's just definining a `u8` constant like we previously had for Sapling in `zcash_primitives` (before the recent refactor). But that doesn't have to block this PR.